internal/config/viper_store.go

```diff
@@ -0,0 +1,221 @@
+// Copyright 2025 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate go tool go.uber.org/mock/mockgen -destination=./mocks.go -package=config github.com/mongodb/mongodb-atlas-cli/atlascli/internal/config ConfigStore
+
+package config
+
+import (
+	"errors"
+	"os"
+	"path/filepath"
+	"slices"
+	"sort"
+	"strings"
+
+	"github.com/pelletier/go-toml"
+	"github.com/spf13/afero"
+	"github.com/spf13/viper"
+)
+
+type ViperConfigStore struct {
+	viper     viper.Viper
+	configDir string
+	fs        afero.Fs
+}
+
+// / ViperConfigStore specific methods
+func LoadViperStore(fs afero.Fs) (*ViperConfigStore, error) {
+	configDir, err := CLIConfigHome()
+	if err != nil {
+		return nil, err
+	}
+
+	v := viper.New()
+
+	v.SetConfigName("config")
+
+	if hasMongoCLIEnvVars() {
+		v.SetEnvKeyReplacer(strings.NewReplacer(AtlasCLIEnvPrefix, MongoCLIEnvPrefix))
+	}
+
+	v.SetConfigType(configType)
+	v.SetConfigPermissions(configPerm)
+	v.AddConfigPath(configDir)
+	v.SetFs(fs)
+
+	v.SetEnvPrefix(AtlasCLIEnvPrefix)
+	v.AutomaticEnv()
+
+	// aliases only work for a config file, this won't work for env variables
+	v.RegisterAlias(baseURL, OpsManagerURLField)
+
+	// If a config file is found, read it in.
+	if err := v.ReadInConfig(); err != nil {
+		// ignore if it doesn't exists
+		var e viper.ConfigFileNotFoundError
+		if !errors.As(err, &e) {
+			return nil, err
+		}
+	}
+	return &ViperConfigStore{
+		viper:     *v,
+		configDir: configDir,
+		fs:        fs,
+	}, nil
+}
+
+func hasMongoCLIEnvVars() bool {
+	envVars := os.Environ()
+	for _, v := range envVars {
+		if strings.HasPrefix(v, MongoCLIEnvPrefix) {
+			return true
+		}
+	}
+
+	return false
+}
+
+func ViperConfigStoreFilename(configDir string) string {
+	return filepath.Join(configDir, "config.toml")
+}
+
+func (s *ViperConfigStore) Filename() string {
+	return ViperConfigStoreFilename(s.configDir)
+}
+
+/// ConfigStore implementation
+
+func (s *ViperConfigStore) Save() error {
+	exists, err := afero.DirExists(s.fs, s.configDir)
+	if err != nil {
+		return err
+	}
+	if !exists {
+		if err := s.fs.MkdirAll(s.configDir, defaultPermissions); err != nil {
+			return err
+		}
+	}
+
+	return s.viper.WriteConfigAs(s.Filename())
```

Comment:
I think is more readable but feel free to ignore 
```suggestion
	if exists {
               return s.viper.WriteConfigAs(s.Filename())
	}

	   return s.fs.MkdirAll(s.configDir, defaultPermissions)
```

Reply:
Will create a ticket to address the actual logic, right now it's just a copy pasted